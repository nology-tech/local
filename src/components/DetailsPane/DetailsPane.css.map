{"version":3,"sources":["../../styles/_variables.scss","DetailsPane.scss","DetailsPane.css"],"names":[],"mappings":"AAOA,6DAAA;AAuBA,qDAAA;AC7BA;EACA,eAAA;EACA,MAAA;EACA,OAAA;EACA,WAAA;EACA,YAAA;EACA,aAAA;EACA,mBAAA;EACA,uBAAA;EACA,oCAAA;ACEA;;ADAA;EACI,WAAA;EACA,YAAA;EACA,oCAAA;EACA,uBAAA;EACA,qBAAA;EACA,aAAA;EACA,sBAAA;EACA,aAAA;ACGJ;;ADDA;EACI,aAAA;ACIJ;;ADFA;EACI,oCAAA;EACA,oCAAA;EACA,WAAA;ACKJ;;ADHA;EACI,aAAA;EACA,uBAAA;ACMJ;;ADJA;EACI,kCDVkB;ECWlB,qBAAA;ACOJ","file":"DetailsPane.css"}