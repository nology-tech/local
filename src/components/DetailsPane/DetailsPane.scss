@use "../../styles/variables.scss" as *;
.details-pane{
position: fixed;
top: 0;
left: 0;
width: 100%;
height: 100%;
display: flex;
align-items: center;
justify-content: center;
background-color: rgba(0, 128, 0, 0);
}
.details-pane__container{
    width: 96vw;
    height: 96vh;
    background-color: rgb(255, 255, 255);
    border: solid 3px black;
    border-radius: 1.5rem;
    display: flex;
    flex-direction: column;
    padding: 1rem;
}
.details-pane__header{
    display: flex;
}
.details-pane__save-btn-container{
    display: flex;
    justify-content: center;
}