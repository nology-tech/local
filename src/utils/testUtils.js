import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
