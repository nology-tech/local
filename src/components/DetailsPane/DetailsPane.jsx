import "./DetailsPane.scss";
import Button from "../Button/Button";

const DetailsPane = ({ onClick, card, buttonName, buttonText }) => {
  return (
    <div className="details-pane" key={card.id}>
      <div className="details-pane__container">
        <div className="details-pane__header">
          <h2>{card.name}</h2>
          <Button buttonText="âœ–" buttonName="close" onClick={onClick} />
        </div>
        <p>{card.description}</p>
        <p>{card.weekdayOpening}</p>
        <p>{card.weekendOpening}</p>
        <p>{card.address}</p>
        <p>{card.price}</p>
        <p>{card.image}</p>
        <div className="details-pane__save-btn-container">
          <Button
            buttonText={buttonText}
            buttonName={buttonName}
            page="favourites"
          />
        </div>
      </div>
    </div>
  );
};

export default DetailsPane;
