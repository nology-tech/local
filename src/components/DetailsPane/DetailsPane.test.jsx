import DetailsPane from "./DetailsPane";
import { customRender } from "../../utils/testUtils";

it("should render the details pane on the display", () => {
  const testCard = {
    id: "0",
    name: "Smith and Sons",
    description: "A restaurant",
    weekdayOpening: "2",
    weedendOpening: "1",
    address: "A street",
    price: "Â£",
    image: "insert here",
  };
  const { container } = customRender(<DetailsPane card={testCard} />);

  expect(container).toMatchSnapshot();
});
