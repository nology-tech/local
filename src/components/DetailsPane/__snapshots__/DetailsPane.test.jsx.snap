// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the details pane on the display 1`] = `
<div>
  <div
    class="details-pane"
  >
    <div
      class="details-pane__container"
    >
      <div
        class="details-pane__header"
      >
        <h2>
          Smith and Sons
        </h2>
        <button
          class="button button--close"
        >
          ✖
        </button>
      </div>
      <p>
        A restaurant
      </p>
      <p>
        2
      </p>
      <p />
      <p>
        A street
      </p>
      <p>
        £
      </p>
      <p>
        insert here
      </p>
      <div
        class="details-pane__save-btn-container"
      >
        <button
          class="button button--undefined"
        />
      </div>
    </div>
  </div>
</div>
`;
